bundler/lib/dependabot/bundler/package_manager.rb

```diff
@@ -40,13 +39,13 @@ def initialize(version)
       sig { override.returns(T::Array[Dependabot::Version]) }
       attr_reader :supported_versions
 
-      sig { override.returns(T::Boolean) }
-      def deprecated?
-        deprecated_versions.include?(version)
-      end
       sig { override.returns(T::Boolean) }
       def unsupported?
-        !deprecated? && version < supported_versions.first
+        # Check if the feature flag for Bundler v1 unsupported error is enabled.
+        return false unless Dependabot::Experiments.enabled?(:bundler_v1_unsupported_error)
+
+        # Determine if the Bundler version is unsupported.
+        supported_versions.any? && version < supported_versions.first
```

Comment:
💅 What do you think of something like 👇?

```suggestion
        supported_versions.all? { |supported| supported > version }
```